### ISSUE TITLE
Checksum payloads

### ISSUE THREAD

**Issue by rjoleary** [oreboot/oreboot#162]
The checksum would check the integrity of the payload and print an error if it doesn't match. This isn't for security.

**Comment by fotonick** [comment]
At what stage are you proposing to checksum? At what stage are you proposing to enforce?

**Comment by rjoleary** [comment]
I imagine it would be in the payloader stage. You can see how coreboot does it here: https://github.com/coreboot/coreboot/blob/master/Documentation/cbfs.txt#L211

**Comment by fotonick** [comment]
Here's my current understanding of what needs to be done:

1. At build time, insert a header before the bytes of the main payload. This would be in `layoutflash`.
2. At run time, verify the checksum against header bytes. This would be in `payload.rs`.

Does that sound right?

**Comment by rjoleary** [comment]
Yes, that sounds right. A simple CRC checksum would do (ex: https://crates.io/crates/crc). Nothing cryptographic is needed.
